core/src/test/scala/unit/kafka/cluster/PartitionTest.scala

```diff
@@ -1257,20 +1250,18 @@ class PartitionTest extends AbstractPartitionTest {
 
     // On initialization, the replica is considered caught up and should not be removed
     partition.maybeShrinkIsr()
-    assertEquals(Set(brokerId, remoteBrokerId), partition.inSyncReplicaIds)
+    assertEquals(Set(brokerId, remoteBrokerId), partition.isrState.isr)
 
     // If enough time passes without a fetch update, the ISR should shrink
     time.sleep(partition.replicaLagTimeMaxMs + 1)
-    val updatedLeaderAndIsr = LeaderAndIsr(
-      leader = brokerId,
-      leaderEpoch = leaderEpoch,
-      isr = List(brokerId),
-      zkVersion = 1)
-    when(stateStore.shrinkIsr(controllerEpoch, updatedLeaderAndIsr)).thenReturn(Some(2))
 
+    // Shrink the ISR
     partition.maybeShrinkIsr()
-    assertEquals(Set(brokerId), partition.inSyncReplicaIds)
-    assertEquals(10L, partition.localLogOrException.highWatermark)
+    assertEquals(alterIsrManager.isrUpdates.size, 1)
```

Comment:
I may have missed it, but do we have tests which verify error handling? I see tests which verify requests get sent, but at a quick glance I didn't see tests of responses.